graphql/schemabuilder/pagination.go

```diff
@@ -393,39 +396,33 @@ func (c *connectionContext) pagesFromEdges(edges []Edge, limit int) (pages []str
 	return pages
 }
 
-func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]interface{}, error) {
-	if args.FilterText == nil || *args.FilterText == "" {
-		return nodes, nil
-	}
-
-	nodesToKeep := make([]bool, len(nodes))
+type SafeBatchNodesToKeep struct {
+	nodesToKeep []bool
+	mux         sync.Mutex
+}
 
+func (c *connectionContext) applyBatchTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs, batchedFields map[string]*graphql.Field) ([]bool, error) {
 	g, ctx := errgroup.WithContext(ctx)
-	for unscopedI, unscopedNode := range nodes {
-		i, node := unscopedI, unscopedNode
+	matchStrings := filter.GetMatchStrings(*args.FilterText)
+	s := SafeBatchNodesToKeep{nodesToKeep: make([]bool, len(nodes), len(nodes))}
```

Comment:
Instead of this wrapper object, can we just inline the lock and nodesToKeep object?

Reply:
also, only need `make([]bool, len(nodes))`